src/test/java/org/jabref/logic/importer/fetcher/CollectionOfComputerScienceBibliographiesFetcherTest.java

```diff
@@ -0,0 +1,64 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.testutils.category.FetcherTest;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@FetcherTest
+class CollectionOfComputerScienceBibliographiesFetcherTest {
+    private CollectionOfComputerScienceBibliographiesFetcher fetcher;
+
+    @BeforeEach
+    public void setUp() {
+        ImportFormatPreferences importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
+        when(importFormatPreferences.getKeywordSeparator()).thenReturn(',');
+        fetcher = new CollectionOfComputerScienceBibliographiesFetcher(importFormatPreferences);
+    }
+
+    @Test
+    public void getNameReturnsCorrectName() {
+        assertEquals("Collection of Computer Science Bibliographies", fetcher.getName());
+    }
+
+    @Test
+    public void getUrlForQueryReturnsCorrectUrl() throws MalformedURLException, URISyntaxException, FetcherException {
+        String query = "java jdk";
+        URL url = fetcher.getURLForQuery(query);
+        assertEquals("http://liinwww.ira.uka.de/bibliography/rss?query=java+jdk&sort=score", url.toString());
+    }
+
+    @Test
+    public void performSearchReturnsMatchingMultipleEntries() throws FetcherException {
+        List<BibEntry> searchResult = fetcher.performSearch("jabref");
+        BibEntry bibEntry = searchResult.get(0);
+        assertNotNull(bibEntry.getField(StandardField.ABSTRACT));
```

Comment:
On the one hand, I think, the aim is to be careful when testing and be open to a new format. On the other hand, JabRef decided to test for the concrete format returned by the site. Seeing changes there very often, leads to an early attention of us (and not users showing up complaining that a fetcher does not work; refs https://github.com/JabRef/jabref/issues/6369)

See https://github.com/JabRef/jabref/blob/master/src/test/java/org/jabref/logic/importer/fetcher/INSPIREFetcherTest.java for some examples.

Could you please add these kinds of tests, too?

Reply:
I have changed the tests so that we now check explicitly that each field is as we expect.